src/simulation/ApplyLoad.cpp

```diff
@@ -0,0 +1,266 @@
+
+    LedgerKey contractCodeLedgerKey;
+    contractCodeLedgerKey.type(CONTRACT_CODE);
+    contractCodeLedgerKey.contractCode().hash = sha256(wasmBytes);
+
+    mLoadCodeKey = contractCodeLedgerKey;
+
+    auto const& lm = mApp.getLedgerManager();
+    auto uploadTx = mTxGenerator.createUploadWasmTransaction(
+        lm.getLastClosedLedgerNum() + 1, 0, wasmBytes, contractCodeLedgerKey,
+        std::nullopt);
+
+    closeLedger({uploadTx.second});
+
+    for (auto kvp : mTxGenerator.getAccounts())
+    {
+        auto salt = sha256("Load contract " + std::to_string(kvp.first));
+
+        auto createTx = mTxGenerator.createContractTransaction(
+            lm.getLastClosedLedgerNum() + 1, 0, contractCodeLedgerKey,
+            wasmBytes.size() + 160, salt, std::nullopt);
+        closeLedger({createTx.second});
+
+        auto instanceKey =
+            createTx.second->sorobanResources().footprint.readWrite.back();
+
+        TxGenerator::ContractInstance instance;
+        instance.readOnlyKeys.emplace_back(mLoadCodeKey);
+        instance.readOnlyKeys.emplace_back(instanceKey);
+        instance.contractID = instanceKey.contractData().contract;
+        mLoadInstances.emplace(kvp.first, instance);
+    }
+}
+
+void
+ApplyLoad::benchmark()
+{
+    auto& lm = mApp.getLedgerManager();
+    std::vector<TransactionFrameBasePtr> txs;
+
+    auto resources = multiplyByDouble(lm.maxLedgerResources(true),
+                                      TRANSACTION_QUEUE_SIZE_MULTIPLIER);
+
+    auto const& accounts = mTxGenerator.getAccounts();
+    std::vector<uint64_t> shuffledAccounts(accounts.size());
+    std::iota(shuffledAccounts.begin(), shuffledAccounts.end(), 0);
+    stellar::shuffle(std::begin(shuffledAccounts), std::end(shuffledAccounts),
+                     gRandomEngine);
+
+    for (auto accountIndex : shuffledAccounts)
+    {
+        auto it = accounts.find(accountIndex);
+        releaseAssert(it != accounts.end());
+
+        auto instanceIter = mLoadInstances.find(it->first);
+        releaseAssert(instanceIter != mLoadInstances.end());
+        auto const& instance = instanceIter->second;
+        auto tx = mTxGenerator.invokeSorobanLoadTransaction(
+            lm.getLastClosedLedgerNum() + 1, it->first, instance,
+            rust_bridge::get_write_bytes().data.size() + 160, std::nullopt);
+
+        if (!anyGreater(tx.second->getResources(false), resources))
+        {
+            resources -= tx.second->getResources(false);
+        }
+        else
+        {
+            for (size_t i = static_cast<size_t>(Resource::Type::OPERATIONS);
+                 i <= static_cast<size_t>(Resource::Type::WRITE_LEDGER_ENTRIES);
+                 ++i)
+            {
+                auto type = static_cast<Resource::Type>(i);
+                if (tx.second->getResources(false).getVal(type) >
+                    resources.getVal(type))
+                {
+                    CLOG_INFO(Perf, "Ledger {} limit hit during tx generation",
```

Comment:
This probably should be a metric as we would want to see if the distribution is skewed towards a certain metric. 
Actually a more useful thing to look at would be to compute the utilization percent for every resource type. If we have a resource constantly underutilized, then we may need to do something about the test configuration (unless the skew was intentional).

Reply:
Seems like this has been addressed.